### ISSUE TITLE
Remote REPL

### ISSUE THREAD

**Issue by leviska** [zed-industries/zed#15196]
### Check for existing issues

- [X] Completed

### Describe the feature

Run jupyter repl on remote machines

From what I could understand from code:
* Currently `repl_store` is global, and not updated when changing projects (i.e. connecting to remote project)
* `JupyterSettings` are global too, and project settings are ignored (there is a big change that remote machine needs different kernel configs)

It seems that current code _could_ work on remote, if it weren't global, and it would update to changing workspaces and connecting to remote.

**Comment by leviska** [comment]
I was trying to do this myself, but got a little bit stuck due to lack of knowledge around code base:
* I could get my local config to match remote config, but then it would actually read local jupyter kernelspec instead of remote. I guess that's because I first open local workspace, then remote, and the global `repl_store` is initialized with local fs, not remote
* I don't quite understand the difference between project/workspace/editor and it's relation to each other (see next)
* As far as I understand, we need to `observe_new_views` inside `ReplStore::init`, and create a new repl for each (workspace/editor)
* I was trying to look at `file_finder` to understand, and it works with workspaces, but here we also need editor too to get the code that we want to run, so I got a little stuck with a problem of matching both states correctly. It seems we need to have only one of (editor/workspace), and get the other from that, but I don't get it what the "parent/child" relationship here if it even exists

**Comment by pipegalera** [comment]
Same problem here: 
<img width="518" alt="Screenshot" src="https://github.com/user-attachments/assets/888b74bc-c4a1-470d-a173-97a1a5b650b2">

**Comment by KorigamiK** [comment]
I am also facing this issue and adding a default kernel selection for the remote project in its settings.json also doesn't seem to fix this issue

**Comment by rgbkrk** [comment]
Confusingly, I'm working on support for a remote jupyter server which is _not_ the same as the kernels via remoting (over SSH). I'll need some verbiage to differentiate between these.

**Comment by zachcp** [comment]
This would be a very useful feature.  Thanks again for a great editor.

**Comment by bleaktwig** [comment]
+1. This would be a very nice feature to have when working remotely.

**Comment by lunw1024** [comment]
Is this feature being actively worked on? Is there an ETA?

**Comment by wrenger** [comment]
> it seems like the recent zed versions don't even have the option for repl in remote projects

Seems to be intentional as of #20397
But I don't know why it's also disabled for ssh remotes, there is code for remote kernels...

**Comment by KorigamiK** [comment]
> > it seems like the recent zed versions don't even have the option for repl in remote projects
> 
> Seems to be intentional as of [#20397](https://github.com/zed-industries/zed/pull/20397) But I don't know why it's also disabled for ssh remotes, there is code for remote kernels...

yeah, this sounds like they wanted to focus on other things first but this shouldn't be very time talking to fix either

**Comment by zxy198717** [comment]
any updates?

**Comment by stk-code** [comment]
It might be a user friendly to state that in the REPL / or Remote documentation.  This would have saved me some time and despair over my incompetence.

**Comment by 10hith** [comment]
Trying to use Zed on windows using WSL2 integration and this has become a blocker for me. 
The Jupyter kernel on wsl2 is not detected; This happens to be the single reason I am not moving to zed completely. Will stay here for updates.

**Comment by rgbkrk** [comment]
> Trying to use Zed on windows using WSL2 integration and this has become a blocker for me.
The Jupyter kernel on wsl2 is not detected

I don't have a Windows machine to diagnose it but I'm guessing we're looking for Jupyter kernels in App Data https://github.com/runtimed/runtimed/blob/a7bec5d805bae26650ccbbac9baaa7946d43844b/crates/runtimelib/src/dirs.rs#L88. What's the path to your kernels on WSL?
